lang/py/src/avro/io.py

```diff
@@ -475,6 +565,47 @@ def write_crc32(self, bytes):
     """
     self.write(STRUCT_CRC32.pack(crc32(bytes) & 0xffffffff));
 
+  def write_date_int(self, datum):
+    """
+    Encode python date object as int
+    int stores the number of days from
+    the unix epoch, 1 January 1970 (ISO calendar).
+    """
+    delta_date = datum - datetime.date(1970, 1, 1)
+    self.write_int(delta_date.days)
+
+  def write_time_millis_int(self, datum):
+    """
+    Encode python time object as int
+    int stores the number of milliseconds from midnight, 00:00:00.000
+    """
+    delta_time = datum.hour*3600000 + datum.minute * 60000 + datum.second * 1000 + datum.microsecond / 1000
+    self.write_int(int(delta_time))
+
+  def write_time_micros_long(self, datum):
+    """
+    Encode python time object as long
+    long stores the number of microseconds from midnight, 00:00:00.000000
+    """
+    delta_time = datum.hour*3600000000 + datum.minute * 60000000 + datum.second * 1000000 + datum.microsecond
+    self.write_long(long(delta_time))
+
+  def write_ts_millis_long(self, datum):
+    """
+    Encode python time object as int
+    int stores the number of milliseconds from midnight, 00:00:00.000
+    """
+    delta_time = (datum - datetime.datetime(1970, 1, 1, 0, 0, 0, 0)).total_seconds() * 1000;
```

Comment:
there are packages like simplejson that are supported for lower versions of pyhton. I have replaced  `total_seconds` with [this](http://stackoverflow.com/a/28089673)